### ISSUE TITLE
`bytes`/`string` literal concatenation

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#13661]
In some languages subsequent string literals are concatenated prior to compilation, which proves useful for containing large amounts of verbatim text in source code, while splitting it across multiple lines or logical boundaries.

One good example for Solidity is NFT metadata-JSON and SVG generation, where text is built up programatically. Current best practice is use `string.concat` and break logical parts.

Here's an excerpt from [Uniswap NFT](https://github.com/Uniswap/v3-periphery/blob/main/contracts/libraries/NFTSVG.sol):
```solidity
                '<path id="text-path-a" d="M40 12 H250 A28 28 0 0 1 278 40 V460 A28 28 0 0 1 250 488 H40 A28 28 0 0 1 12 460 V40 A28 28 0 0 1 40 12 z" />',
                '<path id="minimap" d="M234 444C234 457.949 242.21 463 253 463" />',
                '<filter id="top-region-blur"><feGaussianBlur in="SourceGraphic" stdDeviation="24" /></filter>',
                '<linearGradient id="grad-up" x1="1" x2="0" y1="1" y2="0"><stop offset="0.0" stop-color="white" stop-opacity="1" />',
                '<stop offset=".9" stop-color="white" stop-opacity="0" /></linearGradient>',
                '<linearGradient id="grad-down" x1="0" x2="1" y1="0" y2="1"><stop offset="0.0" stop-color="white" stop-opacity="1" /><stop offset="0.9" stop-color="white" stop-opacity="0" /></linearGradient>',
                '<mask id="fade-up" maskContentUnits="objectBoundingBox"><rect width="1" height="1" fill="url(#grad-up)" /></mask>',
                '<mask id="fade-down" maskContentUnits="objectBoundingBox"><rect width="1" height="1" fill="url(#grad-down)" /></mask>',
                '<mask id="none" maskContentUnits="objectBoundingBox"><rect width="1" height="1" fill="white" /></mask>',
                '<linearGradient id="grad-symbol"><stop offset="0.7" stop-color="white" stop-opacity="1" /><stop offset=".95" stop-color="white" stop-opacity="0" /></linearGradient>',
                '<mask id="fade-symbol" maskContentUnits="userSpaceOnUse"><rect width="290px" height="200px" fill="url(#grad-symbol)" /></mask></defs>',
                '<g clip-path="url(#corners)">',
```

This results in a bunch of items on the stack and a lot of more runtime overhead.

It would be useful allowing `'hello' 'world'` to be compounded as `helloworld`.

**Comment by hrkrshnn** [comment]
Random comment: we should also be moving these string into `code` and use `codecopy`, rather than `mstore(...)` with 32-byte chunks.

**Comment by cameel** [comment]
Isn't it already implemented? #7524 has been merged ages ago.

Parser already has this covered unless I'm misunderstanding the feature being requested here: https://github.com/ethereum/solidity/blob/e29a68d37421ba46364e2e0d839ffa49a8625257/libsolidity/parsing/Parser.cpp#L2029-L2039
